library/init/category/reader.lean

```diff
@@ -0,0 +1,80 @@
+/-
+Copyright (c) 2017 Microsoft Corporation. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Sebastian Ullrich
+
+The reader monad transformer for passing immutable state.
+-/
+
+prelude
+import init.category.lift init.category.id
+universes u v w
+
+structure reader_t (r : Type u) (m : Type u → Type v) (α : Type u) : Type (max u v) :=
+(run : r → m α)
+
```

Comment:
At `state.lean`, you have `attribute [pp_using_anonymous_constructor] state_t`.
Should we also have 
```
attribute [pp_using_anonymous_constructor] reader_t
```
here?

Reply:
Probably :) . Fixed.